virtool/users/data.py

```diff
@@ -161,39 +170,98 @@ async def create(
+                    legacy_id=document["_id"],
+                    handle=handle,
+                    password=document["password"] if password else None,
+                    force_reset=force_reset,
+                    last_password_change=now,
+                    b2c_display_name=b2c_user_attributes.display_name
+                    if b2c_user_attributes
+                    else "",
+                    b2c_given_name=b2c_user_attributes.given_name
+                    if b2c_user_attributes
+                    else "",
+                    b2c_family_name=b2c_user_attributes.family_name
+                    if b2c_user_attributes
+                    else "",
+                    b2c_oid=b2c_user_attributes.oid if b2c_user_attributes else "",
+                )
+            )
+
         return await self.get(document["_id"])
 
+    async def delete(self, legacy_id: str):
+        """
+        Delete the user with the matching legacy_id.
+
+         :param legacy_id: legacy id of the user to delete
+         :return:
+        """
+        async with both_transactions(self._mongo, self._pg) as (mongo, pg):
+            await pg.execute(delete(SQLUser).where(SQLUser.legacy_id == legacy_id))
+            await self._mongo.users.delete_one({"_id": legacy_id}, session=mongo)
+
+    async def delete_all(self):
+        """
+        Delete all users.
+         :return:
+        """
+        async with both_transactions(self._mongo, self._pg) as (mongo, pg):
+            await pg.execute(delete(SQLUser))
+            await self._mongo.users.delete_many({}, session=mongo)
+
     async def create_first(self, handle: str, password: str) -> User:
         """
         Create the first instance user.
 
         :param handle: the user handle
         :param password: the password
-        :return:
+        :return: the user created
         """
         if await self.check_users_exist():
             raise ResourceConflictError("Virtool already has at least one user")
 
         if handle == "virtool":
             raise ResourceConflictError("Reserved user name: virtool")
 
-        async with self._mongo.create_session() as session:
+        async with both_transactions(self._mongo, self._pg) as (mongo, pg):
+            now = virtool.utils.timestamp()
+
             document = await create_user(
-                self._mongo, handle, password, False, session=session
+                self._mongo, handle, password, False, session=mongo
             )
-
             await self._mongo.users.update_one(
                 {"_id": document["_id"]},
                 {"$set": {"administrator": True}},
-                session=session,
+                session=mongo,
+            )
+            user = SQLUser(
+                legacy_id=document["_id"],
+                handle=handle,
+                password=document["password"],
+                force_reset=False,
+                last_password_change=now,
             )
+            pg.add(user)
```

Comment:
nitpick: I think you only use `user` here once so I would just define it as it is used